chiselFrontend/src/main/scala/chisel3/core/package.scala

```diff
@@ -55,14 +55,17 @@ package chisel3 {
     implicit class fromStringToLiteral(val str: String) {
       /** String to UInt parse, recommended style for constants.
         */
-      def U(): UInt = UInt.Lit(parse(str), parsedWidth(str))  // scalastyle:ignore method.name
+      def U(): UInt = str.asUInt() // scalastyle:ignore method.name
       /** String to UInt parse with specified width, recommended style for constants.
         */
-      def U(width: Width): UInt = UInt.Lit(parse(str), width)  // scalastyle:ignore method.name
+      def U(width: Width): UInt = str.asUInt(width) // scalastyle:ignore method.name
 
       /** String to UInt parse, recommended style for variables.
         */
-      def asUInt(): UInt = UInt.Lit(parse(str), parsedWidth(str))
+      def asUInt(): UInt = {
+        val bigInt = parse(str)
+        UInt.Lit(bigInt, Width(bigInt.bitLength))
```

Comment:
This causes DecodeSpec failures. The (suppressed) message is:

      Message: requirement failed: The literal value 0 was elaborated with a specificed width of 0 bits, but at least 1 bits are required.

and this is because `bigInt.bitLength` is reporting a value of 0 for `0`. According to [BigInt](http://www.scala-lang.org/api/current/scala/math/BigInt.html) bitLength "Returns the number of bits in the minimal two's-complement representation of this BigInt, excluding a sign bit."

The line should be:

        UInt.Lit(bigInt, Width(math.max(bigInt.bitLength, 1)))